datalake/backend/tests/test_import_dataset.py

```diff
@@ -0,0 +1,139 @@
+        "body": {"message": "Bad Request: 'metadata_url' is a required property"},
+    }
+
+
+def test_should_return_required_property_error_when_missing_dataset_id() -> None:
+    # When
+    body = {}
+    body["metadata_url"] = any_s3_url()
+    body["version_id"] = any_dataset_version_id()
+    response = lambda_handler(body, any_lambda_context())
+
+    assert response == {
+        "statusCode": 400,
+        "body": {"message": "Bad Request: 'dataset_id' is a required property"},
+    }
+
+
+def test_should_return_required_property_error_when_missing_version_id() -> None:
+    # When
+    body = {}
+    body["dataset_id"] = any_dataset_id()
+    body["metadata_url"] = any_s3_url()
+    response = lambda_handler(body, any_lambda_context())
+
+    assert response == {
+        "statusCode": 400,
+        "body": {"message": "Bad Request: 'version_id' is a required property"},
+    }
+
+
+@mark.timeout(timedelta(minutes=20).total_seconds())
+@mark.infrastructure
+def test_should_batch_copy_files_to_storage(
+    s3_client: S3Client,
+    s3_control_client: S3ControlClient,
+    sts_client: STSClient,
+) -> None:
+    # pylint: disable=too-many-locals
+    # Given a metadata file with an asset
+    first_asset_content = any_file_contents()
+    first_asset_multihash = sha256(first_asset_content).hexdigest()
+
+    dataset_id = any_dataset_id()
+    version_id = any_dataset_version_id()
+
+    with S3Object(
+        BytesIO(initial_bytes=first_asset_content),
+        ResourceName.DATASET_STAGING_BUCKET_NAME.value,
+        any_safe_filename(),
+    ) as asset_s3_object:
+        metadata_stac_object = deepcopy(MINIMAL_VALID_STAC_OBJECT)
+        metadata_stac_object["assets"] = {
+            any_stac_asset_name(): {
+                "href": asset_s3_object.url,
+                "checksum:multihash": first_asset_multihash,
+            },
+        }
+        metadata_content = dumps(metadata_stac_object).encode()
+        with S3Object(
+            BytesIO(initial_bytes=metadata_content),
+            ResourceName.DATASET_STAGING_BUCKET_NAME.value,
+            any_safe_filename(),
+        ) as metadata_s3_object:
+            asset_id = f"DATASET#{dataset_id}#VERSION#{version_id}"
+
+            with ProcessingAsset(
+                asset_id=asset_id, multihash=None, url=metadata_s3_object.url
+            ) as metadata_processing_asset, ProcessingAsset(
+                asset_id=asset_id,
+                multihash=first_asset_multihash,
+                url=asset_s3_object.url,
+            ) as processing_asset:
+
+                # When
+                body = {}
+                body["dataset_id"] = dataset_id
+                body["version_id"] = version_id
+                body["metadata_url"] = metadata_s3_object.url
+                body["type"] = any_valid_dataset_type()
+
+                response = lambda_handler(body, any_lambda_context())
+
+                final_states = ["Complete", "Failed", "Cancelled"]
```

Comment:
No constant values in boto3  available to us for this. Do you mean creating an Enum?

Reply:
No, I was hoping that the Python API code would expose constants. Otherwise we're hardcoding values which belong to *their* API, which should be avoided whenever possible since those values could change.